lib/spack/llnl/util/symlink.py

```diff
@@ -296,15 +306,15 @@ def _windows_read_hard_link(link: str) -> str:
         return links.pop()
     elif len(links) > 1:
         # TODO: How best to handle the case where 3 or more paths point to a single inode?
-        raise SymlinkError(f"Found multiple paths pointing to the same inode {links}")
+        raise SymlinkError(f"Found multiple paths pointing to the same inode {links}", errno=17)
     else:
-        raise SymlinkError("Cannot determine hard link source path.")
+        raise SymlinkError("Cannot determine hard link source path.", errno=18)
```

Comment:
(if the error codes stay - see other comments first) Use `SymlinkErrorType.NoHardlinkSourcePath`